document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -244,17 +247,29 @@ protected String parseNonCompositeFilter(Filter filter, Params.Builder paramsBui
         return filterString.append("(").append(collect).append(")").toString();
       case CONTAINS:
         // TODO: Matches condition inside an array of documents
-      case EXISTS:
-        // TODO: Checks if key exists
+        throw new UnsupportedOperationException("Operation is not supported.");
       case NOT_EXISTS:
-        // TODO: Checks if key does not exist
-      case NEQ:
+        // Postgres jsonb doesnt have a separate operator for NOT EXISTS
+        // A 'NOT' can be used in the filter for exists to get the same behaviour.
+        isNotOperator = true;
+      case EXISTS:
```

Comment:
exists and not_exists filters are working on keyspace, checks if a key exists in documents or not.
```
{"id":"testKey1","name":"abc1","size":-10.2,"isCostly":false,"_id":"default:testKey1"}
{"id":"testKey2","name":"abc2","size":10.4,"isCostly":false,"_id":"default:testKey2"}
{"id":"testKey3","name":"abc3","size":30,"isCostly":false,"city":"bangalore","_id":"default:testKey3"}
{"id":"testKey4","name":"abc4","size":30,"isCostly":false,"city":null,"_id":"default:testKey4"}
```
MogoQuery: For exists filter
```
db.mytest.find( { city: { $exists: true } } ) -> returns documents iwth ids ['testKey3', 'testKey4']
```
MogoQuery: For non-exists filter
```
db.mytest.find( { city: { $exists: false } } ) -> returns documents iwth ids ['testKey1', 'testKey2']
```
Currently, for this PR, we are getting a below query expression for the `exists` filter for postgres which is throwing error
```
SELECT * FROM mytest WHERE CAST (document->>'city' AS BOOLEAN) ? 'TRUE'
```
**Proposal:**
Postgres: exists filter
```
SELECT * FROM mytest WHERE document->'city' IS NOT null
```
Postgres: non-exists filter
```
SELECT * FROM mytest WHERE document->'city' IS null
```

In case, if you want to try out sample data in Postgres (creates table & insert data)

```
CREATE TABLE mytest (id TEXT PRIMARY KEY,document jsonb NOT NULL,created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW())
INSERT INTO mytest (id,document) VALUES( 'default:testKey1', '{"id":"testKey1","name":"abc1","size":-10.2,"isCostly":false,"_id":"default:testKey1"}' :: jsonb) ON CONFLICT(id) DO UPDATE SET document = '{"id":"testKey1","name":"abc1","size":-10.2,"isCostly":false,"_id":"default:testKey1"}'::jsonb 
INSERT INTO mytest (id,document) VALUES( 'default:testKey2', '{"id":"testKey2","name":"abc2","size":10.4,"isCostly":false,"_id":"default:testKey2"}' :: jsonb) ON CONFLICT(id) DO UPDATE SET document = '{"id":"testKey2","name":"abc2","size":10.4,"isCostly":false,"_id":"default:testKey2"}'::jsonb 
INSERT INTO mytest (id,document) VALUES( 'default:testKey3', '{"id":"testKey3","name":"abc3","size":30,"isCostly":false,"city":"bangalore","_id":"default:testKey3"}' :: jsonb) ON CONFLICT(id) DO UPDATE SET document = '{"id":"testKey3","name":"abc3","size":30,"isCostly":false,"city":"bangalore","_id":"default:testKey3"}'::jsonb 
INSERT INTO mytest (id,document) VALUES( 'default:testKey4', '{"id":"testKey4","name":"abc4","size":30,"isCostly":false,"city":null,"_id":"default:testKey4"}' :: jsonb) ON CONFLICT(id) DO UPDATE SET document = '{"id":"testKey4","name":"abc4","size":30,"isCostly":false,"city":null,"_id":"default:testKey4"}'::jsonb 
```

I think we also have to refactor a bit our current implementation `parseNonCompositeFilter` which works with value space with operator `->>`.

Reply:
Fixed the implementation to not perform `CAST(` for EXISTS op

We cannot use
`SELECT * FROM mytest WHERE document->'city' IS NOT null`

for Postgres: exists filter

since it will evaluate to False if the key exists but its value is null
this behaviour will be different than mongo's `$exist`operator 

On the other hand postgres jsonb's `?` operator has exactly same behaviour as mongo's
and also a bit faster since data of the field is not accessed.

Reply:
@utk-spartan As discussed offline, I was talking of using key-space operator (field accessor check) - `document->'city'` to use.